### ISSUE TITLE
React Native - Cannot read property 'version' of undefined

### ISSUE THREAD

**Issue by zachrnolan** [plaid/link#310]
### A detailed description of the steps to reproduce the issue

* Followed install instructions
* Run project from Xcode
* Error: `Cannot read property of 'version' of undefined`

### What you expected to see?

A working app. It seems like the instructions are missing some linking steps?

### What you saw instead?

Red screen with error: `Cannot read property of 'version' of undefined`

**Comment by afh** [comment]
Hi @zachrnolan :wave:,

thanks for reaching out and for using Plaid Link with React Native.

Does your application embed `LinkKit.framework` and link against `LinkKit.framework` and `libRNLinkkit.a` (see screenshot below)?

If it does could you provide a minimal project with which the issue you run into is reproducible?

<img width="50%" alt="react-native" src="https://user-images.githubusercontent.com/16507/54644367-55e19200-4a56-11e9-8d7e-b0d91be45415.png">

**Comment by zachrnolan** [comment]
Hi @afh,

Yes, I've added LinkKit.framework to both sections. However, the libRNLinkkit.a is showing up as red in the Libraries folder:

![Slack_-_Astra](https://user-images.githubusercontent.com/66614/54647327-431f8b00-4a5f-11e9-9cc9-73d45db06721.jpg)

I can't seem to add it to the Link Libraries and Frameworks section.

**Comment by zachrnolan** [comment]
I just noticed that libRNLinkkit.a only seems to be referenced in your code, but not in the actual repo that the instructions say to clone.

**Comment by afh** [comment]
Xcode displays unbuilt products in red once their built the are shown in black:

<img width="253" alt="image" src="https://user-images.githubusercontent.com/16507/54651169-db713c00-4a6e-11e9-9ca3-8dfdb445e381.png">

Can your share a minimal example with which the issue you are observing can be reproduced reliably?

**Comment by zachrnolan** [comment]
@afh so I tried following the instructions again and cloned the entire plaid/link repo (the instructions are very unclear) into my own (since this library isn't on NPM). I then tried to follow the instructions.

Here is the minimal example repo: https://github.com/AstraFinance/plaid-link-example

**Comment by afh** [comment]
Thank you for your helpful feedback that the instructions are very unclear.
I'm working on improving the documentation, any suggestions that you'd like make to ensure the documentation is clearer are welcome.

TL;DR to get react-native-plaid-link working in your project:

* Use the following command to make `libRNLinkkit.a` available to your project
`(cd $PATH_TO_YOUR_REACT_NATIVE_PROJECT;yarn add file:$PATH_TO_YOUR_CLONE_OF_THIS_REPO/react-native/linkkit; yarn install; react-native link react-native-plaid-link)`
* Embed `LinkKit.framework` into your application. How you manage the LinkKit.framework on the filesystem  is up to you, you can drag and drop the framework and have Xcode copy the necessary files, symlink it, rsync it, use Carthage or Cocoapods.

Does this resolve the issues you are having with react-native-plaid-link?

<details><summary>Detailed information on the issue</summary>

The documentation has the following issues that will be addressed:

1. When using yarn to add the react-native-plaid-link component a wrong file path is used:
~(cd $PATH_TO_YOUR_REACT_NATIVE_PROJECT;yarn add file:$PATH_TO_YOUR_CLONE_OF_THIS_REPO/react-native-plaid-link; yarn install)~
(cd $PATH_TO_YOUR_REACT_NATIVE_PROJECT;yarn add file:$PATH_TO_YOUR_CLONE_OF_THIS_REPO/react-native/linkkit; yarn install)
2. An important step is missing from that list of commands, which will link the libRNLinkkit.a library to your project:
(cd $PATH_TO_YOUR_REACT_NATIVE_PROJECT;yarn add file:$PATH_TO_YOUR_CLONE_OF_THIS_REPO/react-native/linkkit; yarn install; react-native link react-native-plaid-link)
3. The documentation should be more clear that the LinkKit.frameworks must be integrated into your react-native iOS project

</details>

**Comment by zachrnolan** [comment]
@afh Thanks for the response. I ran the `react-native link react-native-plaid-link` command and I'm still receiving this error:

![RNPlaidLinkExample_xcodeproj](https://user-images.githubusercontent.com/66614/54734214-46983c80-4b5b-11e9-8569-37dc022cf816.jpg)

![RNPlaidLinkExample_xcodeproj](https://user-images.githubusercontent.com/66614/54734218-4ef07780-4b5b-11e9-9d48-a72a93101c87.jpg)

![RNPlaidLinkExample_xcodeproj](https://user-images.githubusercontent.com/66614/54734248-82cb9d00-4b5b-11e9-9d43-e02dc146a36b.jpg)

Are you seeing anything wrong with my setup on the minimal repo I sent over?

**Comment by afh** [comment]
Hi @zachrnolan,
the LinkKit.framework isn't found during compilation of your project, because the **Framework Search Paths** build setting has not been modified to include the path to LinkKit.framework, e.g.: `FRAMEWORK_SEARCH_PATHS = $(SRCROOT)/../link/ios`
<img width="50%" src="https://user-images.githubusercontent.com/16507/54800968-cab4f780-4c21-11e9-839c-1562160fb18c.png">

With that change (and setting a proper public_key in `App.js`) I am able to successfully build and run your example project—thanks for providing it, it has been very useful—and open Plaid Link from it.

I recommend going over the [Plaid Link for iOS integration guide](https://plaid.com/docs/link/ios/)in detail and pay extra attention to the required ["Add custom Run Script step"](https://plaid.com/docs/link/ios/#add-run-script), failing to do so will very likely get your application rejected during App Store submission or review.

Any feedback on which parts of the documentation are unclear or on how to improve them is welcome.

Please close this issue after having verified that with the above notes you are able to successfully integrate Plaid Link for iOS using the React Native Link Bridge.

And thanks for using Plaid Link! 🙂

**Comment by zachrnolan** [comment]
@afh Thank you, I was able to get this working with the framework search path. One more question, for the run script, what should the path look like? I see a POD_ROOT mentioned, however, we don't use cocoapods. (installed it manually)

**Comment by afh** [comment]
Glad to hear that you got this working! 👍 

The script can handle manual integration *and* integration via CocoaPods (hence the use of `$POD_ROOT`).
By default the script assumes it will find the `LinkKit.framework` in the Xcode project folder (`$PROJECT_DIR`). For your [example](https://github.com/AstraFinance/plaid-link-example) the following should be appropriate:

```
LINK_ROOT=${PODS_ROOT:+$PODS_ROOT/Plaid/ios}
cp "${LINK_ROOT:-$PROJECT_DIR}"/../link/ios/LinkKit.framework/prepare_for_distribution.sh "${CODESIGNING_FOLDER_PATH}"/Frameworks/LinkKit.framework/prepare_for_distribution.sh
"${CODESIGNING_FOLDER_PATH}"/Frameworks/LinkKit.framework/prepare_for_distribution.sh
```

**Comment by afh** [comment]
Closing this issue since it hasn't been updated. Seems things are working for you know @zachrnolan. If not please open a new issue.

Thanks for using Plaid and the React Native Link Bridge. 🙂
